how-to/customize-workspace/client/src/framework/platform/platform-override.ts

```diff
@@ -292,6 +296,36 @@ export const overrideCallback: WorkspacePlatformOverrideCallback = async (Worksp
 				return super.quit(payload, callerIdentity);
 			}
 		}
+
+		public async createWindow(
+			options: OpenFin.PlatformWindowCreationOptions,
+			identity?: OpenFin.Identity
+		): Promise<OpenFin.Window> {
+			const window = await super.createWindow(options, identity);
+
+			try {
+				const platform = getCurrentSync();
+				const browserWindow = platform.Browser.wrapSync(window.identity);
+				await updateBrowserWindowButtonsColorScheme(browserWindow);
+			} catch {
+				// Probably not a browser window
+			}
+
+			return window;
+		}
+
+		public async setSelectedScheme(schemeType: ColorSchemeOptionType) {
+			// The color scheme has been updated, so update the theme
+			await setCurrentColorSchemeMode(schemeType);
```

Comment:
rather than adding more and more logic here over time...could we not update the theme.ts via setCurrentColorScheme and that triggers a lifecycle event of theme changed? That way dock, and browser could register an action against that lifecycle event and if a platform dev wanted to hook into they could define a module and register it against the event as well?

Reply:
Yes we could certainly do it that way as well

Reply:
that way if we have a custom button it just cares about telling theme.ts to switch and the lifecycle event will trigger everything else (rather than all the logic being in the override).